### ISSUE TITLE
frontends.openid_connect._create_provider needs token_endpoint to silence Provider constructor

### ISSUE THREAD

**Issue by mrvanes** [IdentityPython/SATOSA#69]
When "code" is enabled as response_types_supported in OpenIDConnectFrontend configuration, Provider refuses to instantiate due to missing required token_endpoint. This endpoint will be added later in code where response_types_supported is evaluated (#171) but then it's too late. Adding the following line in frontend/openid_connect.py@74 works around the exception.

```
            "token_endpoint": True
```

It's a quick'n'dirty fix but silences the Provider constructor, while eventually creating a correct token_endpoint after completing register_endpoints() anyway.

**Comment by mrvanes** [comment]
This one is better. It leaves token_endpoint untouched when not needed.
```
            "scopes_supported": scopes_supported
        }
+
+        if any("code" in rts for rts in response_types_supported):
+            capabilities["token_endpoint"] = True
```

**Comment by c00kiemon5ter** [comment]
This has been fixed by 8f567583e61142107e6c0b17aa0e94a56f895935
